CONTRIBUTING.rst

```diff
@@ -0,0 +1,177 @@
+.. _developers-contributing:
+
+Contribute to WarpX
+===================
+
+We welcome new contributors!
+Here is how to participate to the WarpX development.
+
+Git workflow
+------------
+
+The WarpX project uses `git <https://git-scm.com>`_ for version control.
+If you are new to git, you can follow one of these tutorials:
+
+- `Learn git with bitbucket <https://www.atlassian.com/git/tutorials/learn-git-with-bitbucket-cloud>`_
+- `git - the simple guide <http://rogerdudler.github.io/git-guide/>`_
+
+Make your own fork and create a branch on it
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The basic WarpX workflow is:
+
+- Fork the main repo (or update it if you already created it).
+- Implement your changes and push them on a new branch ``<branch_name>`` on your fork.
+- Create a Pull Request from branch ``<branch_name>`` on your fork to branch `master` on the main WarpX repo.
```

Comment:
Not sure if too much information for real beginners, but I definitely think it is useful to include in the workflow:
```suggestion
- Run the scripts `.github/workflows/source/hasEOLwhiteSpace`; `.github/workflows/source/hasNonASCII` and `.github/workflows/source/hasTabs` (in the shell you can do that by typing `bash <name_of_script` inside the WarpX repo main folder)
- Use git to compare the changes you did in branch ``<branch_name>`` on your fork to branch `master` on the main WarpX repo
- Create a Pull Request.
```

Reply:
I am note sure to include this right now and here, we might rather do an extra section about local testing in the developer docs in a follow-up (style and unit tests).